### PR TITLE
Fix T5 graph API numerical tests with accurate C++ references

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Reverts the unacceptable ~0.8–1.0 relative-error tolerances and fixes the T5 test reference generator so all `tests_graph_model_t5_*` numerical checks pass at **2e-5**.

## Root cause

The previous fix masked failures by loosening tolerances instead of aligning references with the C++ graph API. Two concrete bugs in `generate_test_data.py` caused O(1) errors:

1. **Weight decoding** — `_weight_from_fortran` treated NNTile flat storage as Fortran-ordered NumPy, scrambling matrices used in PyTorch matmuls. References must invert `fortran_order` via `ravel("C").reshape(shape, order="F")`.
2. **Mask decoding** — causal/cross attention masks stored with `fortran_order` were passed to `_pt_sdpa_eager` without the same inversion, breaking masked attention, decoder blocks, and full model tests.

## Fix

- Implement C++-accurate reference ops: RMSNorm on axis 0 of `(D,S,B)`, FF transpose + GatedMlp path, encoder/decoder blocks, model, and conditional generation.
- Add `_nntile_to_math_array`, `_weight_for_matmul`, and `_mask_from_fortran` helpers.
- Restore uniform **2e-5** forward/backward tolerances for all T5 blocks.
- Remove HuggingFace forward shortcuts that do not match the graph layout.

## Verification

```bash
ctest -R '^tests_graph_model_t5_' --output-on-failure
# 16/16 passed
```

<div><a href="https://cursor.com/agents/bc-e7bae5b5-217c-4ae7-838f-7199f41d4f43"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-e7bae5b5-217c-4ae7-838f-7199f41d4f43"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor push 8ad177103f
